flocker/ca/_ca.py

```diff
@@ -3,7 +3,25 @@
 """
 Low-level logic for a certificate authority.
 
-Uses RSA 4096-bit + SHA 256.
+We have three categories of certificates:
+
+1. Control service, used by AMP and REST API servers. Needs to be
+   validated over HTTP by 3rd party clients, as well as by AMP clients.
+2. Node agents, used by AMP clients in agents. Needs to encode a node
+   UUID, and is validated by the control service.
+3. API clients. Used by HTTP API clients to authenticate, so the control
+   service REST API needs to validate them.
+
+None of these should be able to impersonate the others. We therefore use
+the following Distinguised Name scheme:
+
+1. Control service: common name is "control-service", subjectAltName is
+   administrator-specifiable DNS hostname, to support standard HTTPS
+   client authentication, and extendedKeyUsage is set to "serverAuth".
+2. Node agents: common name is "node-<uuid>", and extendedKeyUsage is set
+   to "clientAuth" (technically clientAuth means HTTPS, but whatevs).
+3. API clients: Common name is set to "user-<username>", and
+   extendedKeyUsage is set to "clientAuth".
```

Comment:
The correct thing for us to do is probably create a new extension type and use that.  This is probably a lot of work and apart from being less abusive of the x509v3 standard (with has interop consequences) I'm not sure it actually makes the software better (rather, if it makes it better, I'm not sure in what ways).  It's perhaps worth a note about follow-up work, though?

:question: